src/components/JsonEditor.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import MonacoEditor, { useMonaco } from '@monaco-editor/react';
+
+interface JsonEditorProps {
+  resource: {};
+  value: string;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+interface RejectionReason {
+  type: string;
+  msg?: string;
+}
+
+const handleUnhandledRejection = (event: PromiseRejectionEvent) => {
+  const reason = event.reason as RejectionReason;
+  if (reason && reason.type === 'cancelation') {
+    console.warn('Operation manually canceled:', reason.msg);
+    event.preventDefault();
+  } else {
+    console.error('Unhandled promise rejection:', reason);
+  }
+};
+
+window.addEventListener('unhandledrejection', handleUnhandledRejection);
```

Comment:
I'm not sure what this `unhandledrejection` is about. Can you add a comment explaining why this is needed?
Also, be careful about writing it here or this handler will be registered every time this file is loaded (probably multiple times when opening different pages)

Reply:
When I use window.addEventListener('unhandledrejection', handleUnhandledRejection); I don't get an error like in the picture 1. However, when I use that part of the code, I get a warning like in picture 2.

Picture 1:
![Screenshot from 2024-08-06 09-53-52](https://github.com/user-attachments/assets/ca6a4c5e-2d1a-4813-a485-c448e77f07d8)
Picture 2:
![Screenshot from 2024-08-06 09-58-19](https://github.com/user-attachments/assets/d039e1c6-8545-4b63-b3f5-111f315b1587)

I think this is the only solution for handling this error. This error will occur only in development mode because the monaco-editor loads twice

Reply:
Then, we will probably need to initialize `monaco` only once. Maybe directly inside the `App` component.
So in the `JsonEditor` component we can forget about the initialization because it was already handled before.
(The `App` is loaded only once, `JsonEditor` can be loaded multiple times, depending on the page that uses it)